### DISCUSSION TITLE
Error documentation

### DISCUSSION THREAD

**Discussion by EmmaTheMartian** [vlang/v#13500] [Ideas]
This is a suggestion to more so V's documentation rather than the language itself. I have been encountering some odd errors recently, in the making of a Linux shell. In this, I've noticed: Both on the Discord, and on the Issue Tracker, my specific error is nowhere to be found.

Due to this, V's documentation should contain a section, or another document which shows every possible error message in V, and what can or will cause it. What do you think?

**Comment by JalonSolov** [comment]
I think it's too early, as the list is changing all the time.  Not as often as the code itself, but...

For example, there are a few error messages I've seen that I really should put in a PR to get the text fixed.

**Reply by EmmaTheMartian** [reply]
Maybe so, although, having some of the more confusing errors documented would make debugging so much easier. I got one error, `non-constant array bound 'string'`, odd thing is, I was using a map, and I was just defining a type.

I looked on the issue tracker and such, not a single person had said anything about the error. I asked on Discord, no responses (That I know of). Therefore, having errors like the example above would make developer's lives easier in the long-run.

**Comment by JalonSolov** [comment]
It would be better to have a Github issue (not a comment on a discussion) asking for a clarification.  That could lead to the error message being improved, which would help everyone.

Just documenting something that's confusing really doesn't help.

**Reply by EmmaTheMartian** [reply]
Okay. I'll make an issue then!
